internal/cmd/rsl/record/record.go

```diff
@@ -12,12 +14,14 @@ type options struct{}
 
 func (o *options) AddFlags(_ *cobra.Command) {}
 
-func (o *options) Run(_ *cobra.Command, args []string) error {
+func (o *options) Run(cmd *cobra.Command, args []string) error {
 	repo, err := repository.LoadRepository()
 	if err != nil {
 		return err
 	}
-
+	if err := repo.VerifyRef(cmd.Context(), args[0], false); err != nil {
```

Comment:
verification relies on the RSL entry, no? IMO, this should create a signed RSL entry (to enable verification of current change), verify, and _remove the RSL entry if verification fails_.

Reply:
Makes sense, I was thinking that we should stop the creation of RSL entries, if their is an invlaid entry already in the RSL

Reply:
I have changed this to revoke the current RSL entry, by skipping it with an annotation entry

Reply:
This was a old comment, at the moment, InvalidateRSLEntry removes the last commit